libs/autofile/autofile.go

```diff
@@ -32,50 +34,70 @@ if err != nil {
 */
 
 const (
-	autoFileOpenDuration = 1000 * time.Millisecond
-	autoFilePerms        = os.FileMode(0600)
+	autoFileClosePeriod = 1000 * time.Millisecond
+	autoFilePerms       = os.FileMode(0600)
 )
 
-// Automatically closes and re-opens file for writing.
+// AutoFile automatically closes and re-opens file for writing. The file is
+// automatically setup to close itself every 1s and upon receiving SIGHUP.
+//
 // This is useful for using a log file with the logrotate tool.
 type AutoFile struct {
-	ID            string
-	Path          string
-	ticker        *time.Ticker
-	tickerStopped chan struct{} // closed when ticker is stopped
-	mtx           sync.Mutex
-	file          *os.File
+	ID   string
+	Path string
+
+	closeTicker       *time.Ticker
+	closeTickerStopCh chan struct{} // closed when closeTicker is stopped
+	hup               chan os.Signal
```

Comment:
`hupc` or `hupCh`?

Reply:
`ch` and `Ch` have a very distinct meaning in our codebase as they represent the reactor communication channels. So I'd lean towards `hupc` and try to use it consistently in the codebase. Attempted that with the recent Transport addition in `p2p`.